### DISCUSSION TITLE
'until' not working as expected, showing day in the future

### DISCUSSION THREAD

**Discussion by amygillies95** [graphite-project/graphite-web#2838] [General]
Hello, 

We use graphite to show order count's of our site. This is currently the graph we have using graphite 0.9.16: 
<img width="1260" alt="Screenshot 2024-06-21 at 11 37 09" src="https://github.com/graphite-project/graphite-web/assets/142903540/11fb6b2e-97ba-4df9-9102-3e4058b5e6b0">

and we are in the middle of upgrading to 1.1.10 using python 3, the graph currently looks like this:
<img width="1267" alt="Screenshot 2024-06-21 at 11 37 16" src="https://github.com/graphite-project/graphite-web/assets/142903540/ccc3b0c4-3a0b-4c55-9fcf-73b6dc88d525">

For some reason, the newer version of 1.1.10 isn't showing past the 'now' for although we have specified +1day in 'until' parameter, as we want it one day ahead. There is no data stored in the database for the Orders_now past now, but we do have the previous data for up to -365 days ago stored. 

The api we are using on both is: 

http://{{host}}/render?width=1250&height=625&from=-2days&until=%2B1d&target=alias(lineWidth(movingAverage(transformNull(order.database.store.count,0),11),3),%22Orders_now%22)&target=alias(movingAverage(timeShift(transformNull(order.database.store.count,0)%2C%227d%22,resetEnd=True),11),%22Orders_7days_ago%22)&target=alias(movingAverage(timeShift(transformNull(order.database.store.count,0)%2C%2228d%22,resetEnd=True),11),%22Orders_28days_ago%22)&target=alias(movingAverage(timeShift(transformNull(order.database.store.count,0)%2C%221345h%22,resetEnd=True),11),%22Orders_56days_ago%22)&target=alias(dashed(color(movingAverage(timeShift(transformNull(order.database.store.count,0)%2C%22364d%22,resetEnd=True),11),%22grey%22)),%22Orders_52w_ago%22)&title=Orders%20Per%205%20minutes

Do you know of any discrepancies or updates between 0.9.16 or 1.1.10, I have looked at the source code and can't seem to see any changes with the until parameter? Or have any idea how I can get this to work as like the first screenshot with the newer version?

Thanks

**Comment by deniszh** [comment]
I'm afraid there are too many code changes happening between 0.9.16 and 1.10.0 and it's very hard to say which exact change caused that.
I was pretty much sure that until > current time never worked in Graphite but looks like I'm wrong and 0.9.16 smh able to do that.
